transaction/DirectedGraph.hpp

```diff
@@ -158,8 +156,8 @@ class DirectedGraph {
   // Class for representing a graph node.
   class DirectedGraphNode {
    public:
-    explicit DirectedGraphNode(transaction_id *data)
-      : data_(data) {}
+    explicit DirectedGraphNode(transaction_id payload)
```

Comment:
Please add `const` for `payload`, as well in many methods in this file.

Also, I suggest to just use `transaction` instead of `payload`.